src/wavm/dynlink.cpp

```diff
@@ -54,10 +70,16 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
 
 WAVM_DEFINE_INTRINSIC_FUNCTION(env, "dlerror", I32, dlerror)
 {
-    faabric::util::getLogger()->debug("S - _dlerror");
+    faabric::util::getLogger()->debug("S - dlerror");
+    std::string errorMessage("Wasm dynamic linking error. See logs");
 
-    // Ignore
-    return 0;
+    wasm::WAVMWasmModule* module = getExecutingWAVMModule();
+    uint32_t wasmStrPtr = module->mmapMemory(errorMessage.size());
+
+    char* strPtr = &Runtime::memoryRef<char>(module->defaultMemory, wasmStrPtr);
+    ::strcpy(strPtr, errorMessage.c_str());
+
+    return wasmStrPtr;
```

Comment:
`dlerror` wasn't actually implemented before, so errors with dynamic linking where ignored (leading to some confusion...)